explorer/src/Pos/Explorer/TestUtil.hs

```diff
@@ -0,0 +1,240 @@
+
+producePureBlock
+    :: (HasConfiguration, HasUpdateConfiguration)
+    => Byte
+    -> BlockHeader
+    -> [TxAux]
+    -> ProxySKBlockInfo
+    -> SlotId
+    -> DlgPayload
+    -> SscPayload
+    -> UpdatePayload
+    -> SecretKey
+    -> Either Text MainBlock
+producePureBlock limit prev txs psk slot dlgPay sscPay usPay sk =
+    createMainBlockPure limit prev psk slot sk $
+    RawPayload txs sscPay dlgPay usPay
+
+leftToCounter :: (ToString s, Testable p) => Either s a -> (a -> p) -> Property
+leftToCounter x c = either (\t -> counterexample (toString t) False) (property . c) x
+
+type BlockNumber   = Word
+type SlotsPerEpoch = Word
+type TotalEpochs   = Word
+
+-- | Function that should generate arbitrary blocks that we can use in tests.
+produceBlocksByBlockNumberAndSlots
+    :: forall m. (HasConfiguration, HasUpdateConfiguration, MonadIO m, Monad m)
+    => BlockNumber
+    -> SlotsPerEpoch
+    -> SlotLeaders
+    -> [SecretKey]
+    -> m [Block]
+produceBlocksByBlockNumberAndSlots blockNumber slotsNumber producedSlotLeaders secretKeys = do
+
+    -- This is just plain wrong and we need to check for it.
+    when (blockNumber < slotsNumber) $ error "Illegal argument."
+
+    let generatedEpochBlocksM :: m [[Block]]
+        generatedEpochBlocksM = forM [0..totalEpochs] $ \currentEpoch -> do
+            generateGenericEpochBlocks Nothing slotsNumber (EpochIndex . fromIntegral $ currentEpoch)
+
+    concat <$> generatedEpochBlocksM
+  where
+
+    totalEpochs :: TotalEpochs
+    totalEpochs = blockNumber `div` slotsNumber
+
+    generateGenericEpochBlocks
+        :: Maybe BlockHeader
+        -> SlotsPerEpoch
+        -> EpochIndex
+        -> m [Block]
+    generateGenericEpochBlocks mBlockHeader slotsPerEpoch epochIndex = do
+        let generatedBlocks = generateEpochBlocks mBlockHeader slotsPerEpoch epochIndex
+
+        let gbToMainBlock :: Block
+            gbToMainBlock = Left . fst $ generatedBlocks
+
+        let mainBlocks :: [MainBlock]
+            mainBlocks = snd generatedBlocks
+
+        let mbToMainBlock :: [Block]
+            mbToMainBlock = Right <$> mainBlocks
+
+        pure $ [gbToMainBlock] ++ mbToMainBlock
+
+    generateEpochBlocks
+        :: Maybe BlockHeader
+        -> SlotsPerEpoch
+        -> EpochIndex
+        -> (GenesisBlock, [MainBlock])
+    generateEpochBlocks mBlockHeader slotsPerEpoch' epochIndex =
+        (epochGenesisBlock, epochBlocks)
+      where
+        epochGenesisBlock :: GenesisBlock
+        epochGenesisBlock = mkGenesisBlock mBlockHeader epochIndex producedSlotLeaders
+
+        epochBlocks :: [MainBlock]
+        epochBlocks =
+            -- TODO(ks): Not correct, but I will fix it later.
+            generateBlocks getPrevBlockHeader <$> blockNumbers
+          where
+            blockNumbers :: [Word]
+            blockNumbers = [1..slotsPerEpoch']
+
+            -- type BlockHeader ssc = Either (GenesisBlockHeader ssc) (MainBlockHeader ssc)
```

Comment:
`ssc` has been removed from `GenesisBlockHeader` and `MainBlockHeader`. Or just remove this comment, it's hard to keep types in a comment up-to-date.

Reply:
Yes, nice catch.